lib/pg_party/model/shared_methods.rb

```diff
@@ -6,13 +6,15 @@ module PgParty
   module Model
     module SharedMethods
       def reset_primary_key
-        if self != base_class
-          base_class.primary_key
-        elsif partition_name = partitions.first
-          in_partition(partition_name).get_primary_key(base_class.name)
-        else
-          get_primary_key(base_class.name)
-        end
+        return base_class.primary_key if self != base_class
+
+        partitions = partitions(include_subpartitions: true)
+        return get_primary_key(base_class.name) unless partitions.any?
```

Comment:
I _think_ `any?` will iterate over the entire array. Maybe we want `partitions.empty?` here?

Reply:
`any?` should return immediately when the first element is found, but I swapped it for `empty?` in https://github.com/webandtech/pg_party/commit/194486cafc08dba50b9e119513eebfa03105435e because it is also more readable